### DISCUSSION TITLE
Error: unable to get baseline for port xxx

### DISCUSSION THREAD

**Discussion by Badajoz95** [microsoft/vcpkg#22827] [Q&A]
Hi, after a few days trying to understand how to implement a simple local example I am stuck on the message "Error: unable to get baseline for port xxx", and I was hoping for some information on how to debug what I need to do please.

I have a set of internal enterprise libraries that I am trying to migrate to vcpkg. I need to support local debugging and hosting in either Azure DevOps GIT Repo or Artifact.

One implementation case is a client project which uses corelibrary. 

### client 
**vcpkg.json**
`{
  "name": "client",
  "version": "1.0.0",
  "description": "client",
  "homepage": "https://dev.azure.com/...",
  "dependencies": [
    {
      "name": "corelibrary",
      "version>=": "2.0.0"
    }
  ]
}`

**vcpkg-configuration.json**
`{
  "default-registry": {
    "kind": "filesystem",
    "path": "..\\C++\\Source",
    "baseline": "2022-01-27"
  }
}`

I'm thinking using filesystem would satisfy the requirement to support local debugging, and once I understand that it should be easy to move to  Azure DevOps GIT Repo or Artifact.

I've read up, nae groomed the documentation at extreme lengths regarding Registries, Versions, etc. I believe I've followed the advice and mirrored the example

### corelibrary

**baseline.json**

`{
  "2022-01-27": {
    "initial": {
      "baseline": "2.0.0",
      "port-version": 0
    }
  }
}`

which finds **\versions\c-\corelibrary..json**

`
{
  "versions": [
    {
      "version": "2.0.0",
      "path": "$/ports/corelibrary/2.0.0"
    }
  ]
}`

which points to **\ports\corelibrary\2.0.0\vcpkg.json**

`{
  "name": "corelibrary",
  "version": "2.0.0",
  "description": "CoreLibrary",
  "homepage": "https://dev.azure.com/...."
}`

Unfortunately, when I run vcpkg install against the client all I get is "Error: unable to get baseline for port corelibrary". If I hack the above configuration files, I can trace that vcpkg is hitting _\ports\corelibrary\2.0.0\vcpkg.json_ but then I am at a loss to understand what next, why the error or how to get more information. There is nothing in buildtrees and I can't see a verbose flag.

Any advice or help would be greatly appreciated please.

**Comment by MarcelNehring** [comment]
@Badajoz95, how do you tell vcpkg where to find your registry?

I had the same error when I forgot to update the `baseline` commit hash of my custom registry in my `vcpkg-configuration.json`. So vcpkg could not find the new lib I added in a later version of the registry.
